iina/ObjcUtils.m

```diff
@@ -9,26 +9,70 @@
 #import <Foundation/Foundation.h>
 #import "ObjcUtils.h"
 
+#import <wchar.h>
+
+#define INDEL_WEIGHT 1
+#define SUBSTITUTION_WEIGHT 4
+
+static inline int min(int a, int b, int c) {
+  int m = a;
+  if (b < m) m = b;
+  if (c < m) m = c;
+  return m;
+}
+
 @implementation ObjcUtils
 
-  + (BOOL)catchException:(void(^)())tryBlock error:(__autoreleasing NSError **)error {
-    @try {
-      tryBlock();
-      return YES;
-    }
-    @catch (NSException *exception) {
-      *error = [[NSError alloc] initWithDomain:exception.name code:0 userInfo:exception.userInfo];
-    }
++ (BOOL)catchException:(void(^)())tryBlock error:(__autoreleasing NSError **)error {
+  @try {
+    tryBlock();
+    return YES;
   }
+  @catch (NSException *exception) {
+    *error = [[NSError alloc] initWithDomain:exception.name code:0 userInfo:exception.userInfo];
+  }
+}
 
-  + (BOOL)silenced:(void(^)())tryBlock {
-    @try {
-      tryBlock();
-      return YES;
-    }
-    @catch (NSException *exception) {
-      // do nothing
-    }
++ (BOOL)silenced:(void(^)())tryBlock {
+  @try {
+    tryBlock();
+    return YES;
+  }
+  @catch (NSException *exception) {
+    // do nothing
   }
+}
+
++ (NSUInteger)levDistance:(NSString *)str0 and:(NSString *)str1 {
```

Comment:
Objective-C (C) is faster, and more convenient when initializing and manipulating fixed 2D arrays like in this algorithm.